src/dbos-executor.ts

```diff
@@ -424,6 +481,21 @@ export class DBOSExecutor implements DBOSExecutorContext {
     this.logger.debug(`Registered communicator ${cfn}`);
   }
 
+  #registerProcedure(ro: MethodRegistrationBase) {
+    const proc = ro.registeredFunction as StoredProcedure<unknown>;
+    const cfn = ro.className + '.' + ro.name;
+
+    if (this.procedureInfoMap.has(cfn)) {
```

Comment:
Does our compiler check for duplicates? Even if we check here, maybe it'd be useful if the compiler throws an error if there are duplicated procedure names, because it can easily cause bugs.